src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -60,7 +64,22 @@ internal class DelegatedCompletionListProvider : CompletionListProvider
             var delegatedRequest = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorCompletionEndpointName, delegatedParams).ConfigureAwait(false);
             var delegatedResponse = await delegatedRequest.Returning<VSInternalCompletionList?>(cancellationToken).ConfigureAwait(false);
 
-            return delegatedResponse;
+            if (delegatedResponse is null)
+            {
+                return null;
+            }
+
+            var rewrittenCompletionList = delegatedResponse;
+            foreach (var rewriter in _responseRewriters)
+            {
+                rewrittenCompletionList = await rewriter.RewriteAsync(rewrittenCompletionList,
```

Comment:
It's hard to say without knowing what an implementation of a rewriter would do, but its kinda odd to me that `rewrittenCompletionList` is mutable, but also that we reassign it to the result. Are implementations expected to create new lists? Or modify the existing one in place? Because the API currently enforces neither of those things. There also isn't an easy way for an implementation to bail out and decide that they don't want to modify anything. For example they can't just return the parameter they were passed, because they might have modified it already. Then again, if there was a way for implementations to return false, they still could have modified the parameter. It seems all just a bit fragile and/or messy.